src/msbuild/.vsts-dotnet-ci.yml

```diff
@@ -98,7 +98,7 @@ jobs:
       publishRunAttachments: true
       mergeTestResults: true
     continueOnError: true
-    condition: eq(variables.onlyDocChanged, 0)
+    condition: and(succeeded(), eq(variables.onlyDocChanged, 0))
```

Comment:
another case of test results publishing behavior change (applies to the rest of the changes in this file)